analyses/survival-analysis/survival-analysis_histology.Rmd

```diff
@@ -0,0 +1,202 @@
+
+## Setup
+
+#### Packages and functions
+
+Read in set up script.
+
+```{r Set up library}
+library(survival)
+library(dplyr)
+library(ggplot2)
+```
+
+#### Set up files and directories
+
+Set up output directories. 
+
+```{r Set up directories}
+data_dir <- file.path("..", "..", "data")
+analysis_dir <- file.path("..", "..", "analyses")
+figure_dir <- file.path("..", "..", "figures")
+
+results_dir <- "results"
+plots_dir <- "plots"
+```
+
+Make output directories.
+
+```{r Make output directories}
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+```
+
+## Declare the direcotry for all the data to be used 
+
+```{r define data path}
+metadata_file <- file.path(data_dir, "pbta-histologies.tsv")
+indep_file <- file.path(data_dir,
+                         "independent-specimens.rnaseq.primary-plus-stranded.tsv")
+tp53_file <- file.path(analysis_dir, 
+                        "tp53_nf1_score/results/tp53_altered_status.tsv")
+# only use stranded since tp53 scores for polyA did not have good accuracy
+tel_stranded_file <- file.path(analysis_dir,
+                         "telomerase-activity-prediction/results/TelomeraseScores_PTBAStranded_FPKM.txt")
+# color palette
+ color_file <- file.path(figure_dir, "palettes/broad_histology_cancer_group_palette.tsv")
+```
+
+## Import the metadata
+
+```{r Read in metadata}
+# read in histologies file
+indep_rna <- readr::read_tsv(indep_file)
+
+# remove cell lines by selecting only solid tissue
+histologies <- readr::read_tsv(metadata_file, guess_max = 10000) %>%
+ dplyr::filter(composition=="Solid Tissue" & tumor_descriptor == "Initial CNS Tumor") %>%
+ select(sample_id, broad_histology, short_histology, cancer_group, molecular_subtype, OS_status, OS_days, PFS_days) %>%
+ distinct()
+
+# read in tp53 scores
+tp53_scores <- readr::read_tsv(tp53_file) %>%
+ select(Kids_First_Biospecimen_ID_DNA, Kids_First_Biospecimen_ID_RNA, sample_id, tp53_score)
+
+# read in telomerase scores
+tel_scores <- readr::read_tsv(tel_stranded_file) %>%
+ mutate(Kids_First_Biospecimen_ID_RNA = SampleID,
+        telomerase_score = NormEXTENDScores) %>%
+ select(Kids_First_Biospecimen_ID_RNA, telomerase_score)
+
+# remove samples without RNA-Seq since both classifiers' results use RNA-Seq
+# independent use primary plus rna-seq samples
+meta_indep <- tp53_scores %>%
+  left_join(tel_scores) %>%
+  left_join(histologies) %>%
+  dplyr::filter(Kids_First_Biospecimen_ID_RNA %in% indep_rna$Kids_First_Biospecimen_ID) %>%
+  # recode the categories
+  dplyr::mutate(OS_status_fixed = case_when(
+    OS_status == "LIVING" ~ 0,
+    OS_status == "DECEASED" ~1
+ ))
+
+```
+ 
+## Cox Regression 
+
+### TP53 scores & Telomerase score - results
+```{r Run Cox regression model for tp53 and telomerase}
+# run the model with different covariates
+for(ind_var in c("tp53_score", "telomerase_score")){
+  model <- paste0("survival::Surv(OS_days, OS_status_fixed) ~ ", ind_var)
```

Comment:
Based on the `survival` docs, the ordering for the arguments is not super clear to me so that's a case for making this more explicit IMO

```suggestion
  model <- paste0("survival::Surv(time = OS_days, event = OS_status_fixed) ~ ", ind_var)
```

That's your intention, correct?

Reply:
yes - corrected.